private/bufpkg/bufmodule/bufmoduleref/util.go

```diff
@@ -20,25 +20,25 @@ import (
 )
 
 // parseModuleReferenceComponents parses and returns the remote, owner, repository,
-// and ref (branch or commit) from the given path.
+// and ref (branch, commit, draft, or tag) from the given path.
 func parseModuleReferenceComponents(path string) (remote string, owner string, repository string, ref string, err error) {
-	remote, owner, rest, err := parseModuleIdentityComponents(path)
-	if err != nil {
-		return "", "", "", "", newInvalidModuleReferenceStringError(path)
-	}
-	restSplit := strings.Split(rest, ":")
-	repository = strings.TrimSpace(restSplit[0])
-	if len(restSplit) == 1 {
-		return remote, owner, repository, "", nil
-	}
-	if len(restSplit) == 2 {
-		ref := strings.TrimSpace(restSplit[1])
+	split := strings.Split(path, ":")
+	switch len(split) {
+	case 1:
+		// path has no colon, no need to handle its ref
+	case 2:
+		ref = strings.TrimSpace(split[1])
 		if ref == "" {
 			return "", "", "", "", newInvalidModuleReferenceStringError(path)
 		}
-		return remote, owner, repository, ref, nil
+	default:
+		return "", "", "", "", newInvalidModuleReferenceStringError(path)
+	}
+	remote, owner, repository, err = parseModuleIdentityComponents(split[0])
```

Comment:
Ah. You're doing some "input cleaning" with the draft/tag before deriving the identity components